salt/fileclient.py

```diff
@@ -696,9 +696,19 @@ def on_header(hdr):
                     # Try to find out what content type encoding is used if
                     # this is a text file
                     write_body[1].parse_line(hdr)  # pylint: disable=no-member
-                    if use_etag and "Etag" in write_body[1]:
+                    # Case insensitive Etag header checking below. Don't break case
+                    # insensitivity unless you really want to mess with people's heads
+                    # in the tests. Note: http.server and apache2 use "Etag" and nginx
+                    # uses "ETag" as the header key. Yay standards!
```

Comment:
The standard says that all HTTP header names are case-insensitive, so it doesn't matter what casing they use.